crates/bevy_ecs/src/component.rs

```diff
@@ -1135,73 +1138,538 @@ impl ComponentCloneBehavior {
     }
 }
 
-/// Stores metadata associated with each kind of [`Component`] in a given [`World`].
+/// Coordinates a registration that is reserved, but not registered.
+trait QueuedComponentRegistration {
+    /// Performs the registration.
+    ///
+    /// # Safery
+    ///
+    /// This must only ever be called once with a unique id.
```

Comment:
This wording was a little unintuitive to me - I initially thought it meant you can call register as long as each call provides a unique id. How about "This must only ever be called once, and must be called with a unique id." Splitting the line up makes it clear these are two separate requirements.